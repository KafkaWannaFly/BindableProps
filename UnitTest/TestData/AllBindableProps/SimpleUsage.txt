using BindableProps;
#if WINDOWS
using Microsoft.UI.Xaml;
#endif
#if __ANDROID__
using Microsoft.UI.Xaml.Controls;
#endif

[AllBindableProps]
public partial class MyClass
{
    // Default field
    string text;

    // Support field with a default value
    string placeHolder = "Do you trust me?";
    
    public TextInput()
    {
        InitializeComponent();
    }
}