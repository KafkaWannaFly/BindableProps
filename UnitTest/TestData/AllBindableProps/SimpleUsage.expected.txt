// <auto-generated/>

namespace global
{
    public partial class MyClass
    {

        public static readonly BindableProperty TextProperty = BindableProperty.Create(
                    nameof(Text),
                    typeof(string),
                    typeof(MyClass),
                    default,
                    propertyChanged: (bindable, oldValue, newValue) =>
                                    ((MyClass)bindable).Text = (string)newValue
                );

        public string Text
        {
            get => text;
            set 
            { 
                OnPropertyChanging(nameof(Text));

                text = value;
                SetValue(MyClass.TextProperty, text);

                OnPropertyChanged(nameof(Text));
            }
        }

        public static readonly BindableProperty PlaceHolderProperty = BindableProperty.Create(
                    nameof(PlaceHolder),
                    typeof(string),
                    typeof(MyClass),
                    "Do you trust me?",
                    propertyChanged: (bindable, oldValue, newValue) =>
                                    ((MyClass)bindable).PlaceHolder = (string)newValue
                );

        public string PlaceHolder
        {
            get => placeHolder;
            set 
            { 
                OnPropertyChanging(nameof(PlaceHolder));

                placeHolder = value;
                SetValue(MyClass.PlaceHolderProperty, placeHolder);

                OnPropertyChanged(nameof(PlaceHolder));
            }
        }

    }
}